src/pal/src/misc/miscpalapi.cpp

```diff
@@ -94,67 +91,142 @@ PAL_GetPALDirectoryW( OUT LPWSTR lpDirectoryName, IN UINT cchDirectoryName )
         /* The path that we return is required to have
            the trailing slash on the end.*/
         lpEndPoint++;
-    }
-    if ( lpFullPathAndName && lpEndPoint && *lpEndPoint != '\0' )
-    {
-        while ( cchDirectoryName - 1 && lpFullPathAndName != lpEndPoint )
-        {
-            *lpDirectoryName = *lpFullPathAndName;
-            lpFullPathAndName++;
-            lpDirectoryName++;
-            cchDirectoryName--;
-        }
-            
-        if ( lpFullPathAndName == lpEndPoint )
-        {
-            *lpDirectoryName = '\0';
-            bRet = TRUE;
-            goto EXIT;
-        }
-        else
+        *lpEndPoint = '\0';
```

Comment:
Moreover, unless I am missing something, it is unnecessary to add the terminator, since you can pass the length to the Set method below. And you don't need to call wcslen to get the length, you can just use `lpEndPoint - lpFullNameAndPath`.